lib/services/user/service.go

```diff
@@ -0,0 +1,92 @@
+package user
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/kennydo/artesia/lib/services"
+	"go.uber.org/zap"
+	"golang.org/x/crypto/bcrypt"
+)
+
+// DBService implements the user service, backed by the DB
+type DBService struct {
+	log *zap.SugaredLogger
```

Comment:
I would prefer we not have a logger here, and instead bubble up errors accordingly. The layer that uses the `DBService` can log the errors that it finds, and we can bubble up detailed errors.

Reply:
Is there a downside to having a logger available? Logging errors can be a bit copy-paste-y and unnnecessary since the caller of these functions can log them, but I am OK with having a pattern for logging successful operations (like successfully creating a new user) since they sometimes have info that isn't returned to the caller.

Reply:
Hm I'm worried about logger explosion (too many things logging to the logger, and forgetting what is saying what), but I may be getting too ahead of myself. If you think this is the right way to go, sounds good to me.